ttnn/cpp/ttnn/operations/conv/conv2d/prepare_conv2d_weights.cpp

```diff
@@ -252,23 +235,27 @@ std::pair<ttnn::Tensor, std::optional<ttnn::Tensor>> prepare_conv_weights_biases
         weight_tensor_ = ttnn::operations::core::to_device(weight_tensor_, device, std::nullopt);
 
     if (bias_tensor.has_value()) {
```

Comment:
Please consider move this block into a small function
```
bias_tensor = prepare_bias_input(bias_tensor, ...whatever_else_params_you_need);
```

The function is already huge and I'd love to see us avoid growing it more and more and instead split it so its easy to follow a high level logic here.